src/cli.rs

```diff
@@ -20,13 +20,13 @@ struct Common {
     /// `bindir`, `datarootdir`, `includedir`, `libdir`
     ///
     /// If they are absolute the prefix is ignored.
-    #[clap(long = "prefix", default_value = "/usr/local")]
+    #[clap(long = "prefix")]
```

Comment:
That would require to use `default_value_if` to make it user friendly or to change the types to Option<T>.

A logic that could be simpler might be:
- Build a Target from [consts](https://doc.rust-lang.org/std/env/consts/index.html#)
- Produce the host defaults and the meson-defaults and set them all using `default_value_if`.
- If --target is present override accordingly.

Reply:
I'm not exactly sure I follow, as your use of `clap::derive` here means that any changes must be doable at build time only. Can you explain it a bit more, once I submit the revised approach?

Reply:
Sure, Parser imples CommandFactory so you can use [command](https://docs.rs/clap/latest/clap/trait.CommandFactory.html#tymethod.command) so you can still extend the cli

Reply:
Ah, then it's definitely not possible (AFAIU).

Currently you obtain Rust's host's information through cargo's `cargo::util::command_prelude::ArgMatchesExt.workspace`. That of course requires instantiating `Command::command()` itself, which is also the instance requiring the default value.